optics_functions/coupling.py

```diff
@@ -0,0 +1,213 @@
+"""
+Coupling
+********
+
+Functions to calculate coupling from twiss dataframes.
+
+"""
+import logging
+from contextlib import suppress
+from typing import Sequence
+
+import numpy as np
+from tfs import TfsDataFrame
+
+from optics_functions.constants import (ALPHA, BETA, GAMMA,
+                                        X, Y, TUNE, DELTA,
+                                        MINIMUM, PI2, PHASE_ADV, S, LENGTH)
+from optics_functions.rdt import calculate_rdts
+from optics_functions.utils import split_complex_columns, timeit
+
+COUPLING_RDTS = ["F1001", "F1010"]
+
+
+LOG = logging.getLogger(__name__)
+
+
+def coupling_from_rdts(df: TfsDataFrame, complex_columns: bool = True, **kwargs):
+    """ Returns the coupling term.
+
+    .. warning::
+        This function changes sign of the real part of the RDTs compared to
+        [FranchiAnalyticFormulas2017]_ to be consistent with the RDT
+        calculations from [CalagaBetatronCoupling2005]_ .
+
+    Args:
+        df (TfsDataFrame): Twiss Dataframe
+        complex_columns (bool): Output complex values in single column of type complex.
+                                If ``False``, split complex columns into
+                                two real-valued columns.
+
+    Keyword Args:
+        **kwargs: Remaining arguments from :func:`~optics_functions.rdt.rdts`
+                  i.e. ``qx``, ``qy``, ``feeddown``, ``loop_phases``
+                  and ``hamiltionian_terms``.
+
+    Returns:
+        New TfsDataFrame with Coupling Columns.
+    """
+    df_res = calculate_rdts(df, rdts=COUPLING_RDTS, **kwargs)
+    for rdt in COUPLING_RDTS:
+        df_res.loc[:, rdt].to_numpy().real *= -1  # definition, also: sets value in dataframe
+
+    if not complex_columns:
+        df_res = split_complex_columns(df_res, COUPLING_RDTS)
+
+    return df_res
+
+
+def coupling_from_cmatrix(df: TfsDataFrame, complex_columns: bool = True,
```

Comment:
yes. And no. Coupling from RDTs is also not from RDTs but from Beta, Alpha, KL and phases ;)
But it's going via the step of c-matrix ... so I left it. Also because it was named that way in metaclass and I guess we are used to this naming